web/src/main/java/org/fao/geonet/Geonetwork.java

```diff
@@ -668,4 +676,31 @@ private DataStore createShapefileDatastore(String indexDir) throws Exception {
         logger.info("NOTE: Using shapefile for spatial index, this can be slow for larger catalogs");
         return ids;
     }
+
+    private void initPasswordEncoder() throws Exception {
+        StandardPBEStringEncryptor encryptor =
+                (StandardPBEStringEncryptor) this._applicationContext.getBean("strongEncryptor");
+
+        String securityPropsPath = ((JeevesApplicationContext) _applicationContext).getServletContext()
+                .getRealPath("WEB-INF/config-security/config-security.properties");
+
+        PropertiesConfiguration conf = new PropertiesConfiguration(securityPropsPath);
+        String encrypterPassword = (String) conf.getProperty("encrypter.password");
+
+        // Creates a random encrypter password if the password has the value 'default'
+        if (encrypterPassword.equals("default")) {
+            encrypterPassword = RandomStringUtils.randomAlphanumeric(10);
+
+            conf.setProperty("encrypter.password", encrypterPassword);
+            conf.save();
```

Comment:
In some case, the webapp folder is readonly. Will this work in such case ? Should we add the capability to (or even move by default) the config file to the data dir ? (same applies for jdbc.properties)